pkg/client/cluster-essentials/clusterinit.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package clusteressentials has APIs for bootstrapping kubernetes cluster
+package clusteressentials
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/client/cluster-essentials/pkg/utils"
+)
+
+type ClusterEssentialsClient struct {
+	ClusterEssentialBundlePath string
+	BundleDestDir              string
+	RestConfig                 *rest.Config
+	BlockingCallEnable         bool
+	Calltimeout                time.Duration
+}
+
+var downloadBundle = utils.DownloadBundle
+
+func PreInstallValidation(restConfig *rest.Config) error {
+	// TODO implementation
+	return nil
+}
+
+func PostInstallValidation(restConfig *rest.Config, blockingEnable bool, timeout time.Duration) error {
+	// TODO implementation
+	return nil
+}
+
+func Install(c ClusterEssentialsClient) error {
```

Comment:
The API consumers will usually use retries to handle transient failures. Let's assume the case of the statement `utils.AppplyResourcesFromManifest` succeeds without error. And assume that while calling `PostInstallValidation` there is some network glitch in reaching the API server and the call fails with an error. If there is a retry loop implemented at consumer side, will the next retry resume `PostInstallValidation`? If not, how do we enforce that?

Reply:
We won't need to have a restart from last failed state, because all operations within `Install` are supposed to be idempotent. More importantly, in this scenario, we should exit at the pre-validation since required resources already exist.

Reply:
Discussed.